src/main/java/codeu/controller/AdminServlet.java

```diff
@@ -0,0 +1,118 @@
+   * server, not when running in a test.
+   */
+
+  @Override
+  public void init() throws ServletException {
+    super.init();
+    setUserStore(UserStore.getInstance());
+    setConversationStore(ConversationStore.getInstance());
+    setMessageStore(MessageStore.getInstance());
+  }
+
+  /**
+   * Sets the UserStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setUserStore(UserStore userStore) {
+    this.userStore = userStore;
+  }
+
+  void setConversationStore(ConversationStore conversationStore) {
+    this.conversationStore = conversationStore;
+  }
+
+  void setMessageStore(MessageStore messageStore) {
+    this.messageStore = messageStore;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+        adminList.add("admin");
+        adminList.add("admin1");
+        adminList.add("admin2");
+        adminList.add("admin3");
+
+      String username = (String)request.getSession().getAttribute("user");
+      if (username == null) {
+        // user is not logged in, don't let them access the admin page
+        response.sendRedirect("/");
+        return;
+      }
+      User user = userStore.getUser(username);
+      if (adminList.contains(username)){
+        user.setIsAdmin(true);
+      }
+      if (user == null) {
+        // user was not found, don't let them access the admin page
+        response.sendRedirect("/");
+        return;
+      }
+      if (user.getIsAdmin() == false) {
+        /* rediects user to homepage if not admin*/
+        response.sendRedirect("/");
+        return;
+      }
+
+      /* an attempt to grab information from the stores to display on the page
+      if the user is admin*/
+          List<User> userList = userStore.getAllUsers();
+          int numOfUsers = userStore.getNumOfUsers();
+          int numOfConvos = conversationStore.getNumOfConversations();
+          int numOfMessages = messageStore.getNumOfMessages();
+
+          request.setAttribute("numOfUsers", numOfUsers);
+          request.setAttribute("numOfMessages", numOfMessages);
+          request.setAttribute("numOfConvos", numOfConvos);
+
+          int mostMessages=0;
+          String newestUser = userList.get(userList.size()-1).getName();
+          String mostActiveUser = "";
+          int numOfAdmins = adminList.size();
+
+          for(int i =0; i< numOfUsers; i++){
+            UUID userId = userList.get(i).getId();
+            int userMessagesListSize = messageStore.getMessagesFromAuthorSize(userId);
+            if (userMessagesListSize > mostMessages){
+              mostMessages = userMessagesListSize;
+              User activeUser = userStore.getUser(userId);
+              mostActiveUser = activeUser.getName();
+            } else if(mostActiveUser == null){
```

Comment:
I think you can get rid of this else if entirely. `mostActiveUser` shouldn't ever be `null`, and we can instead check if it's empty in the JSP and not display anything.